chainstate/test-suite/src/tests/delegation_tests.rs

```diff
@@ -0,0 +1,458 @@
+
+    let tx = TransactionBuilder::new()
+        .add_input(genesis_outpoint.into(), empty_witness(rng))
+        .add_output(TxOutput::CreateStakePool(Box::new(stake_pool_data)))
+        .add_output(TxOutput::Transfer(
+            OutputValue::Coin(amount_to_stake),
+            Destination::AnyoneCanSpend,
+        ))
+        .build();
+    let stake_outpoint = OutPoint::new(OutPointSourceId::Transaction(tx.transaction().get_id()), 0);
+    let transfer_outpoint =
+        OutPoint::new(OutPointSourceId::Transaction(tx.transaction().get_id()), 1);
+
+    tf.make_block_builder().add_transaction(tx).build_and_process().unwrap();
+
+    (pool_id, stake_outpoint, transfer_outpoint)
+}
+
+fn prepare_delegation(
+    rng: &mut (impl Rng + CryptoRng),
+    tf: &mut TestFramework,
+) -> (PoolId, OutPoint, DelegationId, OutPoint, OutPoint) {
+    let (pool_id, stake_outpoint, transfer_outpoint) = prepare_stake_pool(rng, tf);
+
+    let available_amount = get_output_value(
+        utxo::UtxosStorageRead::get_utxo(&tf.storage, &transfer_outpoint)
+            .unwrap()
+            .unwrap()
+            .output(),
+    )
+    .unwrap()
+    .coin_amount()
+    .unwrap();
+
+    let delegation_id = pos_accounting::make_delegation_id(&transfer_outpoint);
+    let tx = TransactionBuilder::new()
+        .add_input(transfer_outpoint.into(), empty_witness(rng))
+        .add_output(TxOutput::CreateDelegationId(
+            Destination::AnyoneCanSpend,
+            pool_id,
+        ))
+        .add_output(TxOutput::Transfer(
+            OutputValue::Coin(available_amount),
+            Destination::AnyoneCanSpend,
+        ))
+        .build();
+    let delegation_outpoint =
+        OutPoint::new(OutPointSourceId::Transaction(tx.transaction().get_id()), 0);
+    let transfer_outpoint =
+        OutPoint::new(OutPointSourceId::Transaction(tx.transaction().get_id()), 1);
+
+    tf.make_block_builder().add_transaction(tx).build_and_process().unwrap();
+
+    (
+        pool_id,
+        stake_outpoint,
+        delegation_id,
+        delegation_outpoint,
+        transfer_outpoint,
+    )
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn create_delegation(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng).build();
+
+        let (pool_id, _, transfer_outpoint) = prepare_stake_pool(&mut rng, &mut tf);
+        let delegation_id = pos_accounting::make_delegation_id(&transfer_outpoint);
+
+        let delegation_spend_destination =
+            Destination::PublicKey(PrivateKey::new_from_rng(&mut rng, KeyKind::Secp256k1Schnorr).1);
+
+        let tx = TransactionBuilder::new()
+            .add_input(transfer_outpoint.into(), empty_witness(&mut rng))
+            .add_output(TxOutput::CreateDelegationId(
+                delegation_spend_destination.clone(),
+                pool_id,
+            ))
```

Comment:
Please add an output for change.

Reply:
why? I don't need it for this test. Or do you suggest adding it as an example of how it works?

Reply:
As proof that change won't cause an issue. I would rather have it with and without.